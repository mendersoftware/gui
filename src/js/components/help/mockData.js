// Copyright 2020 Northern.tech AS
//
//    Licensed under the Apache License, Version 2.0 (the "License");
//    you may not use this file except in compliance with the License.
//    You may obtain a copy of the License at
//
//        http://www.apache.org/licenses/LICENSE-2.0
//
//    Unless required by applicable law or agreed to in writing, software
//    distributed under the License is distributed on an "AS IS" BASIS,
//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//    See the License for the specific language governing permissions and
//    limitations under the License.
export const helpProps = {
  docsVersion: undefined,
  isHosted: false,
  menderVersion: 'master',
  isEnterprise: true,
  tenantCapabilities: {
    hasDeviceConfig: true,
    hasMonitor: true
  },
  tokens: [],
  versions: {
    repos: {
      mender: '3.3.0',
      'mender-artifact': '3.8.0',
      'mender-binary-delta': '1.3.1',
      'mender-cli': '1.8.0',
      'mender-configure-module': '1.0.4',
      'mender-connect': '2.0.2',
      'mender-convert': '3.0.0',
      'mender-gateway': '1.0.0',
      'monitor-client': '1.2.0'
    },
    releaseDate: '2003-01-01'
  }
};
