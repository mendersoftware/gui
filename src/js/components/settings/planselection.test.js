import React from 'react';

import { undefineds } from '../../../../tests/mockData';
import { render } from '../../../../tests/setupTests';
import PlanSelection from './planselection';

describe('PlanSelection component', () => {
  it('renders correctly', () => {
    const { baseElement } = render(<PlanSelection isUpgrade offerValid offerTag="very now" trial updatedPlan="enterprise" />);
    const view = baseElement.firstChild;
    expect(view).toMatchSnapshot();
    expect(view).toEqual(expect.not.stringMatching(undefineds));
  });
});
