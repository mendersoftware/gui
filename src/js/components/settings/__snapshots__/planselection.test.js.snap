// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlanSelection component renders correctly 1`] = `
.emotion-0 {
  border-color: #f7f7f7;
}

.emotion-0.active,
.emotion-0:hover {
  border-color: #d4e9e7;
  box-shadow: 0 1px 6px rgba(0, 0, 0, 0.15);
}

.emotion-0.active {
  background-color: #f7f7f7;
}

.emotion-0.addon {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-column-gap: 10px;
  column-gap: 10px;
  display: grid;
  grid-template-columns: max-content minmax(150px, min-content) 1fr max-content;
  height: initial;
  margin-top: 10px;
  min-width: 550px;
  width: initial;
}

.emotion-0.addon.upgrade {
  grid-template-columns: max-content minmax(120px, min-content) minmax(120px, min-content) 1fr max-content;
}

.emotion-1 {
  margin-bottom: -11px;
  font-size: smaller;
}

.emotion-2 {
  font-size: 1rem;
}

.emotion-3 {
  color: rgba(0, 0, 0, 0.38);
  margin: 15px 0;
}

<div>
  <h3
    class="margin-top"
  >
    Plans
  </h3>
  <div
    class="flexbox space-between"
    style="padding-bottom: 15px;"
  >
    <div
      class="planPanel emotion-0  "
    >
      <div
        class="uppercased align-center muted emotion-1"
      >
        current plan
      </div>
      <h4>
        Basic
         
      </h4>
      <div>
        <div
          class="link-color bold emotion-2"
        >
          $32/month
        </div>
        <div>
          for first 50 devices
        </div>
      </div>
      <ul
        class="unstyled"
      >
        <li>
          <span
            class="emotion-3 "
          >
            Basic OTA features
          </span>
        </li>
      </ul>
    </div>
    <div
      class="planPanel emotion-0  "
    >
      <div
        class="uppercased align-center muted emotion-1"
      >
        upgrade
      </div>
      <h4>
        Professional
         
      </h4>
      <div>
        <div
          class="link-color bold emotion-2"
        >
          $269/month
        </div>
        <div>
          for first 250 devices
        </div>
      </div>
      <ul
        class="unstyled"
      >
        <li>
          <span
            class="emotion-3 "
          >
            + Advanced OTA features
          </span>
        </li>
        <li>
          <span
            class="emotion-3 "
          >
            + Standard support
          </span>
        </li>
      </ul>
    </div>
    <div
      class="planPanel emotion-0 active "
    >
      <div
        class="uppercased align-center muted emotion-1"
      >
        upgrade
      </div>
      <h4>
        Enterprise
         
      </h4>
      <div>
        <div
          class="link-color bold emotion-2"
        >
          Custom pricing
        </div>
        <div>
          unlimited devices
        </div>
      </div>
      <ul
        class="unstyled"
      >
        <li>
          <span
            class="emotion-3 "
          >
            + Enterprise-grade OTA features
          </span>
        </li>
        <li>
          <span
            class="emotion-3 "
          >
            + Premium support
          </span>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
